libraries/c_sdk/standard/mqtt/src/iot_mqtt_serializer_deserializer_wrapper.c

```diff
@@ -623,6 +627,54 @@ IotMqttError_t _IotMqtt_deserializeSubackWrapper( _mqttPacket_t * pSuback )
     /* Deserializing SUBACK packet received from the network. */
     managedMqttStatus = MQTT_DeserializeAck( &pIncomingPacket, &( pSuback->packetIdentifier ), NULL );
     status = convertReturnCode( managedMqttStatus );
+
+    /* Remove rejected subscription as the MQTT LTS library do not remove them, it has to handled in shim */
+    if( status == IOT_MQTT_SERVER_REFUSED )
+    {
+        /* Iterate through each status byte in the SUBACK packet. */
+        for( i = 0; i < remainingLength - sizeof( uint16_t ); i++ )
```

Comment:
Doesn't the server already refuse the subscription? can't we just remove it from the array instead of parsing through the packet data?

Reply:
Not all the subscriptions are refused ,only the subscriptions that are refused will be removed, so we need to parse the SUBACK packet.

Reply:
OK. I see